test/unit/BCoWHelper.t.sol

```diff
@@ -0,0 +1,180 @@
+    helper.tokens(address(pool));
+  }
+
+  function test_TokensRevertWhen_PoolHasMoreThan2Tokens() external {
+    address[] memory invalidTokens = new address[](3);
+    invalidTokens[0] = makeAddr('token0');
+    invalidTokens[1] = makeAddr('token1');
+    invalidTokens[2] = makeAddr('token2');
+    pool.set__tokens(invalidTokens);
+    // it should revert
+    vm.expectRevert(ICOWAMMPoolHelper.PoolDoesNotExist.selector);
+    helper.tokens(address(pool));
+  }
+
+  function test_TokensRevertWhen_PoolTokensHaveDifferentWeights() external {
+    pool.mock_call_getNormalizedWeight(tokens[0], VALID_WEIGHT);
+    pool.mock_call_getNormalizedWeight(tokens[1], VALID_WEIGHT + 1);
+
+    vm.expectRevert(ICOWAMMPoolHelper.PoolDoesNotExist.selector);
+    // it should revert
+    helper.tokens(address(pool));
+  }
+
+  function test_TokensWhenPoolIsSupported() external view {
+    // it should return pool tokens
+    address[] memory returned = helper.tokens(address(pool));
+    assertEq(returned[0], tokens[0]);
+    assertEq(returned[1], tokens[1]);
+  }
+
+  function test_OrderRevertWhen_ThePoolIsNotSupported() external {
+    // it should revert
+    vm.expectRevert(ICOWAMMPoolHelper.PoolDoesNotExist.selector);
+    helper.order(invalidPool, priceVector);
+  }
+
+  function test_OrderWhenThePoolIsSupported(bytes32 domainSeparator) external {
+    // it should query the domain separator from the pool
+    pool.expectCall_SOLUTION_SETTLER_DOMAIN_SEPARATOR();
+    pool.mock_call_SOLUTION_SETTLER_DOMAIN_SEPARATOR(domainSeparator);
+
+    (
+      GPv2Order.Data memory order_,
+      GPv2Interaction.Data[] memory preInteractions,
+      GPv2Interaction.Data[] memory postInteractions,
+      bytes memory sig
+    ) = helper.order(address(pool), priceVector);
+
+    // it should return a valid pool order
+    assertEq(order_.receiver, GPv2Order.RECEIVER_SAME_AS_OWNER);
+    assertLe(order_.validTo, block.timestamp + 5 minutes);
+    assertEq(order_.feeAmount, 0);
+    assertEq(order_.appData, factory.APP_DATA());
+    assertEq(order_.kind, GPv2Order.KIND_SELL);
+    assertEq(order_.buyTokenBalance, GPv2Order.BALANCE_ERC20);
+    assertEq(order_.sellTokenBalance, GPv2Order.BALANCE_ERC20);
+
+    // it should return a commit pre-interaction
+    assertEq(preInteractions.length, 1);
+    assertEq(preInteractions[0].target, address(pool));
+    assertEq(preInteractions[0].value, 0);
+    bytes memory commitment = abi.encodeCall(IBCoWPool.commit, GPv2Order.hash(order_, domainSeparator));
+    assertEq(keccak256(preInteractions[0].callData), keccak256(commitment));
+
+    // it should return an empty post-interaction
+    assertTrue(postInteractions.length == 0);
+
+    // it should return a valid signature
+    bytes memory validSig = abi.encodePacked(pool, abi.encode(order_));
+    assertEq(keccak256(validSig), keccak256(sig));
+  }
+
+  function test_OrderGivenAPriceVector(uint256 priceSkewness, uint256 balanceToken0, uint256 balanceToken1) external {
```

Comment:
I don't really follow _what_ are the `5000, `10_000` and `15_000`, much less _why_ we need them. Do you think it would be clearer if they had names? or some comments?

afaict we will _skew_ the price by a particular factor from current spot price in order to allow for naturally ocurring slippage, is that correct?